python-cim/cim/formatters.py

```diff
@@ -126,4 +126,33 @@ def dump_instance(i, encoding=None, encoding_errors='strict'):
     if encoding:
         instance_str = instance_str.encode(encoding=encoding, errors=encoding_errors).decode(encoding)
 
-    return instance_str
\ No newline at end of file
+    return instance_str
+
+def dump_instances_csv(i, encoding=None, encoding_errors='strict'):
+    """ :type i: ClassInstance """
+
+    cl = i[00].class_layout
+    cd = cl.class_definition
+
+    ret = 'classname,super,key,timestamp1,timestamp2'
+
+    # Get property headers
+    for propname in i[00].properties:
+        ret += ',' + propname
+    ret += '\n'
+
+    #ret += '{0},{1},{2},{3},{4}'.format(cd.class_name, cd.super_class_name, str(i.key), i.ts1, i.ts2)
+
+    for instance in i:
+        ret += '"{0}","{1}","{2}","{3}","{4}"'.format(cd.class_name, cd.super_class_name, str(instance.key), instance.ts1, instance.ts2)
```

Comment:
recommend using the `csv` or `unicodecsv` package to format the rows. it handles escaping properly, which will be useful in the unlikely situation that any of these values contain a quotation mark.